hexrdgui/main_window.py

```diff
@@ -749,21 +749,18 @@ def on_action_edit_apply_hand_drawn_mask_triggered(self):
     def run_apply_hand_drawn_mask(self, dets, line_data):
         if self.image_mode == ViewType.polar:
             for line in line_data:
-                name = unique_name(
-                    HexrdConfig().raw_mask_coords, 'polar_mask_0')
+                name = unique_name(MaskManager().mask_names, 'polar_mask_0')
                 raw_line = convert_polar_to_raw([line])
-                HexrdConfig().raw_mask_coords[name] = raw_line
-                HexrdConfig().visible_masks.append(name)
-                create_polar_mask_from_raw(name, raw_line)
-            HexrdConfig().polar_masks_changed.emit()
+                mask = MaskManager().add_mask(name, raw_line, MaskType.polygon)
+                mask.update_masked_arrays(self.image_mode)
```

Comment:
Can you put `update_masked_arrays` inside `MaskManager().add_mask` so you don't need to call it after every `add_mask` call?

Or, alternatively, you could make a `get_masked_arrays()` function that takes the image mode as an argument, and updates lazily (when the data is requested). Something like this:

```python
def get_masked_arrays(self, image_mode):
    if self.masked_arrays is None or self.masked_arrays_image_mode != image_mode:
        self.update_masked_arrays(image_mode)

    return self.masked_arrays
```

I think it might be a good idea anyways for masks to keep track of which image mode their `masked_arrays` are currently in.

Lazily computing could also prevent repeated computations from happening (in case it were updated several times before the data were retrieved).

Reply:
I do think it's a better approach to perform the update lazily. Lazy updates might save us some computation time (since we might sometimes currently be calling `update_masked_arrays()` when we don't need to, but if we call that function right before retrieving the masked arrays, we are guaranteed to perform it only when we need it).

And then, with the lazy updates, we would need some way to invalidate the `masked_arrays`, probably just by setting it to `None` (or to make it clearer, have a function called `invalidate_masked_arrays()` that does just that). This would force it to be re-computed the next time it is retrieved. And we would do things like that when we, for instance, change the polar pixel resolution size.

I don't think the `MaskManager().view_mode != image_mode` approach would work because the individual masks aren't keeping track of which image mode was used to create its `masked_array`. The mask manager's view mode might not match the view mode that was used to create the mask's `masked_array`. I think those could get out of sync easily.

So I think it would be nice to call a `get_masked_arrays(image_mode)` function that basically checks if its masked array was generated already using that image mode, and just return it if so, or re-generate it if the image mode doesn't match. But if this is too much work for this PR, we can wait on it until sometime in the future...

Reply:
My original question was mostly because I was misunderstanding what you meant at first - you weren't asking to sync the mode on each mask every time we switch image modes, just need to track the mode we were in when we last generated the masks. I think what you've explained makes total sense and I do not think that it's too much for this PR!